### ISSUE TITLE
Query Select not mapped to return type

### ISSUE THREAD

**Issue by maicss** [Automattic/mongoose#12544]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.6.2

### Node.js version

16.17.1

### MongoDB server version

4.9.1

### Typescript version (if applicable)

4.8.4

### Description

![image](https://user-images.githubusercontent.com/9784767/194795565-43aa5f64-3c16-4e8e-9e63-12e302685bd4.png)

### Steps to Reproduce

```
// interface & schema
import { model, Schema, Document } from 'mongoose'

export const DOCUMENT_NAME = 'LoginLog'
export const COLLECTION_NAME = 'loginLog'

export interface ILoginLog extends Document {
  IP: string
  attemptTimes: number
  attemptAccounts: string[]
  bannedTimestamp: number
  banned: boolean
  deleted: boolean
  deletedTimestamp: number
  deletedAccount: string
  adminManuallyDeleteName: string
  adminManuallyDeleted: boolean
  adminManuallyDeleteTimestamp: number
}
const schema = new Schema<ILoginLog>({
  attemptTimes: {type: Number, required: true},
  IP: {type: String, required: true, index: true, unique: true},
  attemptAccounts: {type: [String], required: true},
  bannedTimestamp: {type: Number, required: true},
  deleted: {type: Boolean, required: true},
  deletedTimestamp: {type: Number, required: true},
  deletedAccount: {type: String, required: true},
  adminManuallyDeleteName: {type: String, required: true},
  adminManuallyDeleted: {type: Boolean, required: true},
  adminManuallyDeleteTimestamp: {type: Number, required: true},
}, { timestamps: true, versionKey: false })

export const loginLog = model<ILoginLog>(DOCUMENT_NAME, schema, COLLECTION_NAME)
```

```
// query
loginLog.findOne({ IP }).select('banned -_id').lean<{banned: boolean}>().exec()
```

### Expected Behavior

Should infer return Type to be `<{banned: boolean}>`

**Comment by maicss** [comment]
In method `select`, arg `+banned` should not equal to `banned`?

**Comment by hasezoey** [comment]
> In method `select`, arg `+banned` should not equal to `banned`?

read [`Query.prototype.select()`](https://mongoosejs.com/docs/api/query.html#query_Query-select) to what the symbols (or their absence) means
TL;DR: `-` means that the path is excluded, `+` means it is force included (overwrites what is set with the schema option [`select`](https://mongoosejs.com/docs/api/schematype.html#schematype_SchemaType-select)), and no symbol means it is included (unless overwritten)

**Comment by maicss** [comment]
My Bad. Thanks
